bokehjs/src/coffee/api/typings/models/tools.d.ts

```diff
@@ -78,19 +78,23 @@ declare namespace Bokeh {
     export var BoxSelectTool: { new(attributes?: IBoxSelectTool, options?: ModelOpts): BoxSelectTool };
     export interface BoxSelectTool extends SelectTool, IBoxSelectTool {}
     export interface IBoxSelectTool extends ISelectTool {
+        callback?: Callback | ((source: this) => void);
```

Comment:
Why `this`? Shouldn't this be `DataSource`? If that's the case, perhaps `callback` could be moved to the super class (`SelectTool`). The same should be done on Python side, but there it will conflict with how we do documentation.